app/src/main/res/layout/activity_movie_details.xml

```diff
@@ -0,0 +1,394 @@
+            android:textColor="@color/movie_details_a_storyline_text_color"
+            android:textSize="@dimen/movie_details_a_storyline_text_size"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/storyline_header" />
+
+        <TextView
+            android:id="@+id/cast_header"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="@dimen/margin_4x"
+            android:fontFamily="sans-serif-black"
+            android:gravity="start"
+            android:paddingStart="@dimen/padding_2x"
+            android:paddingEnd="@dimen/padding_2x"
+            android:shadowColor="@color/movie_details_a_cast_header_shadow_color"
+            android:shadowDx="0"
+            android:shadowDy="6"
+            android:shadowRadius="12"
+            android:text="@string/movie_details_a_cast_header"
+            android:textColor="@color/movie_details_a_cast_header_text_color"
+            android:textSize="@dimen/movie_details_a_cast_header_text_size"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/storyline" />
+
+        <androidx.cardview.widget.CardView
+            android:id="@+id/cast_1"
+            android:layout_width="0dp"
+            android:layout_height="0dp"
+            android:layout_marginStart="@dimen/margin_2x"
+            android:layout_marginEnd="@dimen/margin_0_5x"
+            app:cardCornerRadius="@dimen/movie_details_a_cast_corner_radius"
+            app:layout_constraintDimensionRatio="H,1:1"
+            app:layout_constraintEnd_toStartOf="@id/cast_2"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/cast_header">
+
+            <ImageView
+                android:layout_width="match_parent"
+                android:layout_height="match_parent"
+                android:contentDescription="@string/movie_details_a_review_image_content_description"
+                android:scaleType="centerCrop"
+                android:src="@drawable/cast_1" />
+        </androidx.cardview.widget.CardView>
+
+        <TextView
+            android:id="@+id/cast_1_name"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="@dimen/margin_1x"
+            android:fontFamily="sans-serif-black"
+            android:gravity="center_vertical"
+            android:text="@string/movie_details_a_cast_1_name"
+            android:textColor="@color/movie_details_a_cast_name_text_color"
+            android:textSize="@dimen/movie_details_a_cast_name_text_size"
+            app:layout_constraintEnd_toEndOf="@id/cast_1"
+            app:layout_constraintStart_toStartOf="@id/cast_1"
+            app:layout_constraintTop_toBottomOf="@id/cast_1" />
```

Comment:
you know that ConstraintLayout likes when you use chains. what do you think about it?

Reply:
Added layout_constraintBottom_toTopOf="@id/cast_N_name" to all cardView with id = "cast_N"